db/compaction/compaction_picker_fifo.cc

```diff
@@ -83,7 +83,12 @@ Compaction* FIFOCompactionPicker::PickTTLCompaction(
           break;
         }
       }
-      total_size -= f->compensated_file_size;
+      // avoid underflow
+      if (total_size < f->compensated_file_size) {
+        total_size = 0;
+      } else {
+        total_size -= f->compensated_file_size;
```

Comment:
Should it use `file_size`, not `compensated_file_size`, to match `GetTotalFilesSize()`?